### PR TITLE
04 nc news

### PR DESCRIPTION
**Pull request by vaishkhullar**

### CODE REVIEW

**Review by corkesw — COMMENTED** [review]
Hi Vaish, you will need to remove the .only from the test suite and then resubmit this ticket

**Comment by vaishkhullar** [comment]
> Hi Vaish, you will need to remove the .only from the test suite and then resubmit this ticket

Great - done

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          title: expect.any(String),
+          article_id: expect.any(Number),
+          body: expect.any(String),
+          topic: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+        });
+      });
+  });
+  test("receiving a request for an article_id that doesn't exist", () => {
```
Comment (k-huggs) [inline __tests__/app.test.js:81]: Missing a status 400 test for if the end point isn't a valid

File: app.js [hunk 2]
```diff
@@ -1,14 +1,30 @@
 const express = require("express");
-const {} = require("./controllers");
+require("jest-sorted");
+
 const apiRouter = require("./routers/api.router");
 
 const app = express();
+//parsing the body
```
Comment (k-huggs) [inline app.js:7]: delete this line

File: controllers/topics.controllers.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const { getTopics } = require("../modellers/topics.model");
+
```
Comment (k-huggs) [inline controllers/topics.controllers.js:2]: controller is called topics controller but has article functions in it

File: controllers/topics.controllers.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+exports.getTopics = (req, res) => {
+  getTopics().then((topics) => {
+    res.status(200).send(topics);
+  });
+};
+
+exports.getArticles = (req, res, next) => {
+  selectArticles()
+    .then((articles) => {
+      res.status(200).send(articles);
```
Comment (k-huggs) [inline controllers/topics.controllers.js]: send back the articles in an object

File: modellers/articles.model.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    .then(({ rows: [article] }) => {
+      if (article) {
+        return article;
+      } else {
+        return Promise.reject({ status: 404, msg: "not found" });
+      }
+    });
+};
+
+// exports.articleExists = (article_id) => {
```
Comment (k-huggs) [inline modellers/articles.model.js]: delete this code if you're not using it

**Review by k-huggs — COMMENTED** [review]
a few changes to be made:
Navigate to this branch,
make the changes,
add, commit and push 
merge the branch into main,
then pull down this onto your main and onto your local 06 branch (there maybe some lines which cause a small conflict potentially)

**Comment by k-huggs** [comment]
Have a look at the comments :)

File: __tests__/app.test.js (hunk 1 above)
Comment (vaishkhullar) [reply __tests__/app.test.js:81]: sorted thanks

File: app.js (hunk 2 above)
Comment (vaishkhullar) [reply app.js:7]: removed as per comment - thanks

File: controllers/topics.controllers.js (hunk 3 above)
Comment (vaishkhullar) [reply controllers/topics.controllers.js:2]: Created a new article controller and amended code accordingly

File: modellers/articles.model.js (hunk 5 above)
Comment (vaishkhullar) [reply modellers/articles.model.js]: done

**Pull request merged**